addons/mail/static/src/core/common/store_service.js

```diff
@@ -370,13 +370,16 @@ export const storeService = {
     start(env, services) {
         const store = makeStore(env);
         store.discuss = {};
-        store.discuss.activeTab = env.services.ui.isSmall ? "mailbox" : "all";
+        store.discuss.activeTab = "main";
         onChange(store.Thread, "records", () => store.updateBusSubscription());
         services.ui.bus.addEventListener("resize", () => {
             if (!services.ui.isSmall) {
-                store.discuss.activeTab = "all";
+                store.discuss.activeTab = "main";
             } else {
-                store.discuss.activeTab = store.discuss.thread?.type ?? "all";
+                store.discuss.activeTab =
+                    (store.discuss.thread?.type !== "mailbox"
+                        ? store.discuss.thread?.type
+                        : null) ?? "main";
```

Comment:
This pattern is not great, look like a soup of `?` it's hard to predict the content of `activeTab`.

Reply:
I know, I made the simplest fix on this LOC I could, without drastically changing these LOCs more heavily.
Functionally it's correct and fixes the issue, which is a higher priority than "clean" code.
IMO this is more than enough maintainable, so if this takes some effort to "clean" it now then this is wasted time for the time being

Reply:
I disagree, cleaning code is never a wast of time as it avoid technical debt. And here I feel like you are creating technical debt.
I think this is better: 
```suggestion
store.discuss.activeTab = "main";
if (store.discuss.thread?.type && store.discuss.thread?.type !== "mailbox") {
    store.discuss.activeTab = store.discuss.thread.type;
}
```

Reply:
Cleaning code can be good time investment, it could be a huge waste of time and can do more harm than good.
The tricky part is that it's named "clean", enforcing a positive interpretation of this adjective, but in practice it's a ideological movement to impose a way to code that can make the code worse and maintainability poor.

With that in mind, "clean" code is not always good.

You proposed LOCs are IMO relatively equivalent. Both are ok, there are tradeoffs with both of them.
On one end, the statements are simpler with your solution. On the other end, it repeats `store.discuss.` quite a lot, and the 1st statement is a lie as it assigns a wrong `activeTab` that is overrides just after, making the code harder to understand by just reading at bits of it.

Reply:
I believe I converted an hard to read code pattern to a commonly used one making it "clean" because most dev have seen and use it before and understand it in the blink of an eye. 
We are entering bad faith territory now. If you don't see that my proposition is easier to read and have to question yourself about what clean code is, I don't know what to tell you.